### PR TITLE
Live reload of Murmur SSL settings via SIGUSR1 (including runtime certificate-swap for Let's Encrypt)

### PR DESCRIPTION
**Pull request by mkrautz**
This PR contains two commits.

The first one adds a new method to MetaParams called loadSSLSettings().
It loads the settings just as MetaParams::read() would, but it has been refactored to a function.
If the function fails, the settings in MetaParams are left as-is. If the function succeeds, the MetaParams
object is updated with the new settings.
Then, it adds the Meta::reloadSSLSettings() method, which reloads SSL MetaParams's SSL settings and re-initializes SSL for all servers that use MetaParams's certificate/key.
Finally, it hooks up SIGUSR1 to invoke Meta::reloadSSLSettings().

The second commit cleans up MetaParams() by removing the old SSL settings load code. The code is replaced by a call to loadSSLSettings(), and appropriate error handling.

### CODE REVIEW

File: src/murmur/Meta.cpp [hunk 1]
```diff
@@ -574,6 +610,27 @@ Meta::~Meta() {
 #endif
 }
 
+bool Meta::reloadSSLSettings() {
+	// Reload SSL settings.
+	if (Meta::mp.loadSSLSettings()) {
```
Comment (mkrautz) [inline src/murmur/Meta.cpp]: Inverted logic

File: src/murmur/UnixMurmur.cpp [hunk 2]
```diff
@@ -228,7 +228,11 @@ void UnixMurmur::handleSigUsr1() {
 	ssize_t len = ::read(iUsr1Fd[1], &tmp, sizeof(tmp));
 	Q_UNUSED(len);
 
-	qWarning("Received USR1 signal... Ignoring...");
+	if (meta) {
+		qWarning("Meta: trying to reload SSL settings");
+		meta->reloadSSLSettings();
+		qWarning("Meta: done reloading SSL settings");
```
Comment (Kissaki) [inline src/murmur/UnixMurmur.cpp]: Shouldn’t this indicate success or failure? We have a return value with that information.

File: src/murmur/Meta.cpp [hunk 3]
```diff
@@ -365,7 +365,56 @@ void MetaParams::read(QString fname) {
 	bSendVersion = typeCheckedFromSettings("sendversion", bSendVersion);
 	bAllowPing = typeCheckedFromSettings("allowping", bAllowPing);
 
-	qsCiphers = typeCheckedFromSettings("sslCiphers", qsCiphers);
+	if (!loadSSLSettings()) {
+		qFatal("MetaParams: Failed to load SSL settings. See previous errors.")
```
Comment (Kissaki) [inline src/murmur/Meta.cpp]: Missing semicolon

Comment (mkrautz) [inline src/murmur/Meta.cpp]: Fixed.

File: src/murmur/UnixMurmur.cpp (hunk 2 above)
Comment (mkrautz) [inline src/murmur/UnixMurmur.cpp]: Fixed.

**Comment by mkrautz** [comment]
> Inverted logic as you pointed out.

Done

**Comment by mkrautz** [comment]
> Additional/follow-up I'd like: Define dhparams later, right before use. Currently, it's defined way before, in a oneliner with other stuff.

Looking into it.

**Comment by mkrautz** [comment]
>> Additional/follow-up I'd like: Define dhparams later, right before use. Currently, it's defined way before, in a oneliner with other stuff.
>
> Looking into it.

Done

**Comment by mkrautz** [comment]
PTAL.

**Comment by mkrautz** [comment]
I think we should clean up `Server::initializeCert()` before landing this...

(Also, this PR doesn't actually apply the new cert as-is)

**Comment by mkrautz** [comment]
Marked the PR as WIP once again.

**Pull request merged**